### PR TITLE
treat experiment as closed when its target groups are full

### PR DESCRIPTION
**Pull request by bbonf**
I thought we had this already

### CODE REVIEW

File: api/tests.py [hunk 1]
```diff
@@ -223,3 +223,13 @@ def test_upload_fail_without_session(self):
...
+
+    def test_experiment_not_open_when_groups_are_full(self):
+        self.group_a.completion_target = 0
+        self.group_b.completion_target = 0
+        self.group_a.save()
+        self.group_b.save()
+
+        response = self.client.post(reverse('api:participant', args=[self.exp.access_id]))
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response.json()['result'], 'ERR_NOT_OPEN')
```
Comment (tymees) [inline api/tests.py:235]: Why not ``ResponseCodes.ERR_NOT_OPEN``?

File: experiments/models.py [hunk 2]
```diff
@@ -95,7 +95,12 @@ def is_open(self):
         """An experiment is open if it is both approved and set to 'open'.
         While an experiment should not be able to have the status 'open' without being approved,
         we check both to be sure."""
```
Comment (tymees) [inline experiments/models.py:97]: The docstring should be updated as well

File: experiments/models.py [hunk 3]
```diff
@@ -95,7 +95,12 @@ def is_open(self):
...
-        return self.state == self.OPEN and self.approved
+        experiment_open = self.state == self.OPEN and self.approved
+        groups_open = True
+        if self.uses_groups():
+            groups_open = any((group.is_open() for group in self.targetgroup_set.all()))
```
Comment (tymees) [inline experiments/models.py:101]: It might be an idea to refactor this into a ``has_open_groups`` method? I could live without tho ;)

Comment (bbonf) [reply experiments/models.py:101]: I started doing that, but then it seems counterintuitive when `uses_groups()` is false, so I'd rather not have that for now.

Comment (tymees) [reply experiments/models.py:102]: Fair enough

**Review by tymees — APPROVED** [review]

**Pull request merged**